enumerables.rb

```diff
@@ -0,0 +1,145 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index(given_index = nil)
+    return enum_for(:my_each_with_index) unless block_given?
+
+    index = 0
+    index = given_index unless given_index.nil?
+    my_each do |x|
+      yield(x, index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    select_array = []
+    return enum_for(:my_select) unless block_given?
+
+    my_each do |x|
+      select_array << x if yield x
+    end
+    select_array
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |x|
+      return false if reg_exp(pattern, x) || num_exp(pattern, x)
+
+      unless block_given?
+        return false unless x
+
+        return true
+      end
+
+      return false unless yield x
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    my_each do |x|
+      return true if reg_exp(pattern, x) || num_exp(pattern, x)
+
+      unless block_given?
+        return true if x
+      end
+      return true if yield x
+    end
+    false
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

Sample testcase:
```ruby
let(:true_array) { [nil, false, true, []] }
let(:false_array) { [nil, false, nil, false] }

expect(true_array.my_any?).to be true_array.any?
expect(false_array.my_any?).to be false_array.any?
```
Should return `true` if `my_any?` works well